In [2]:
import pandas as pd


## Initial Data Review

In this section, we examine the shape of the dataset and verify whether it contains any missing values or duplicate rows.

In [7]:
# Initial Data Review
df = pd.read_csv("dataset-uci.csv", sep=";")

print("Shape:", df.shape)

display(df.head(20))

print("\nData types:")
display(df.dtypes)

print("\nFeatures:")
print(df.columns.tolist())

print("\nMissing values per column:")
display(df.isnull().sum(axis=0))

print("\nNumber of duplicate rows:")
print(df.duplicated().sum())


Shape: (319, 39)


,Gallstone Status,Age,Gender,Comorbidity,Coronary Artery Disease (CAD),Hypothyroidism,Hyperlipidemia,Diabetes Mellitus (DM),Height,Weight,...,High Density Lipoprotein (HDL),Triglyceride,Aspartat Aminotransferaz (AST),Alanin Aminotransferaz (ALT),Alkaline Phosphatase (ALP),Creatinine,Glomerular Filtration Rate (GFR),C-Reactive Protein (CRP),Hemoglobin (HGB),Vitamin D
0,0,50,0,0,0,0,0,0,185,"92,8",...,40,134,20,22,87,"0,82","112,47",0,16,33
1,0,47,0,1,0,0,0,0,176,"94,5",...,43,103,14,13,46,"0,87","107,1",0,"14,4",25
2,0,61,0,0,0,0,0,0,171,"91,1",...,43,69,18,14,66,"1,25","65,51",0,"16,2","30,2"
3,0,41,0,0,0,0,0,0,168,"67,7",...,59,53,20,12,34,"1,02","94,1",0,"15,4","35,4"
4,0,42,0,0,0,0,0,0,178,"89,6",...,30,326,27,54,71,"0,82","112,47",0,"16,8","40,6"
5,0,96,0,0,0,0,0,0,155,49,...,30,65,13,13,60,"1,46","43,74",0,11,"45,8"
6,0,37,0,0,0,0,0,0,185,"67,1",...,40,60,15,14,72,"0,77","118,25",0,"13,8",20
7,0,41,0,0,0,0,0,0,176,114,...,34,464,26,28,69,"1,3","73,73","0,11","16,5","24,5"
8,0,38,0,0,0,0,0,0,186,"93,7",...,33,679,68,102,70,"0,91","110,63","1,57","16,5","22,7"
9,0,38,0,0,0,0,0,0,171,"68,6",...,43,129,19,34,75,"0,91","110,63",0,"16,6","15,6"



Data types:


Gallstone Status                                  int64
Age                                               int64
Gender                                            int64
Comorbidity                                       int64
Coronary Artery Disease (CAD)                     int64
Hypothyroidism                                    int64
Hyperlipidemia                                    int64
Diabetes Mellitus (DM)                            int64
Height                                            int64
Weight                                              str
Body Mass Index (BMI)                               str
Total Body Water (TBW)                              str
Extracellular Water (ECW)                           str
Intracellular Water (ICW)                           str
Extracellular Fluid/Total Body Water (ECF/TBW)      str
Total Body Fat Ratio (TBFR) (%)                     str
Lean Mass (LM) (%)                                  str
Body Protein Content (Protein) (%)              


Features:
['Gallstone Status', 'Age', 'Gender', 'Comorbidity', 'Coronary Artery Disease (CAD)', 'Hypothyroidism', 'Hyperlipidemia', 'Diabetes Mellitus (DM)', 'Height', 'Weight', 'Body Mass Index (BMI)', 'Total Body Water (TBW)', 'Extracellular Water (ECW)', 'Intracellular Water (ICW)', 'Extracellular Fluid/Total Body Water (ECF/TBW)', 'Total Body Fat Ratio (TBFR) (%)', 'Lean Mass (LM) (%)', 'Body Protein Content (Protein) (%)', 'Visceral Fat Rating (VFR)', 'Bone Mass (BM)', 'Muscle Mass (MM)', 'Obesity (%)', 'Total Fat Content (TFC)', 'Visceral Fat Area (VFA)', 'Visceral Muscle Area (VMA) (Kg)', 'Hepatic Fat Accumulation (HFA)', 'Glucose', 'Total Cholesterol (TC)', 'Low Density Lipoprotein (LDL)', 'High Density Lipoprotein (HDL)', 'Triglyceride', 'Aspartat Aminotransferaz (AST)', 'Alanin Aminotransferaz (ALT)', 'Alkaline Phosphatase (ALP)', 'Creatinine', 'Glomerular Filtration Rate (GFR)', 'C-Reactive Protein (CRP)', 'Hemoglobin (HGB)', 'Vitamin D']

Missing values per column:


Gallstone Status                                  0
Age                                               0
Gender                                            0
Comorbidity                                       0
Coronary Artery Disease (CAD)                     0
Hypothyroidism                                    0
Hyperlipidemia                                    0
Diabetes Mellitus (DM)                            0
Height                                            0
Weight                                            0
Body Mass Index (BMI)                             0
Total Body Water (TBW)                            0
Extracellular Water (ECW)                         0
Intracellular Water (ICW)                         0
Extracellular Fluid/Total Body Water (ECF/TBW)    0
Total Body Fat Ratio (TBFR) (%)                   0
Lean Mass (LM) (%)                                0
Body Protein Content (Protein) (%)                0
Visceral Fat Rating (VFR)                         0
Bone Mass (B


Number of duplicate rows:
0


### Initial Data Review Summary

The dataset contains 319 observations and 39 columns. The features include demographic information, medical conditions, body composition measurements, and laboratory test results.

No missing values were found in the dataset, and no duplicate rows were identified. Therefore, no handling of missing data or removal of duplicate observations is required at this stage.

## Cleaning